index.html

```diff
@@ -207,36 +208,47 @@ <h2>Legend</h2>
        return p;
      }
 
-     function sanitizeHTML(value) {
-       // Only support links and code.
-       const template = document.createElement('template');
-       const div = template.content.ownerDocument.createElement('div');
-       template.content.append(div); // force parser context to "in body"
-       div.innerHTML = value;
-       const iterator = document.createNodeIterator(div);
-       let node = iterator.nextNode(); // This is the div
-       while (node = iterator.nextNode()) {
-         if (node.nodeType === Node.TEXT_NODE) {
-           continue;
+     function parseMarkdown(markdown, enableLinks = true) {
+       // Create a document fragment to hold the parsed elements
+       const fragment = document.createDocumentFragment();
+
+       // Regex for either code or link (only # or https://)
+       const markdownPattern = enableLinks ? /(`([^`]+)`|\[([^\]]+)\]\(((?:#|https:\/\/)[^\)]+)\))/g : /(`([^`]+)`)/g;
```

Comment:
It looks like you have two obvious phases here: link parsing and code parsing.  Could that be two separate passes?  That might remove the need for the argument as well.

```js
function expandLinks(md) {
  // match then expandInline in the text chunks before appending those.
}
function expandInline(md) {
  // match /`([^`]+)`/ and maybe /\b*([^*]+)*\b/ as well.
}
```

Reply:
Separately, you don't need the '/g' if you are parsing in chunks like this.